src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -694,5 +765,43 @@ public FormValidation doCheckLabelString(@QueryParameter String value, @QueryPar
             }
             return FormValidation.ok();
         }
+
+        public ListBoxModel doFillWindowsPasswordCredentialsIdItems(@AncestorInPath Jenkins context, @QueryParameter String value) {
+            if (context == null || !context.hasPermission(Item.CONFIGURE)) {
+                return new StandardListBoxModel();
+            }
+            List<DomainRequirement> domainRequirements = new ArrayList<DomainRequirement>();
+            return new StandardListBoxModel()
+                    .withEmptySelection()
+                    .withMatching(
+                            CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class),
+                            CredentialsProvider.lookupCredentials(
+                                    StandardUsernamePasswordCredentials.class, context, ACL.SYSTEM,
+                                    domainRequirements));
+        }
+
+        public ListBoxModel doFillWindowsPrivateKeyCredentialsIdItems(@AncestorInPath Jenkins context, @QueryParameter String value) {
+            if (context == null || !context.hasPermission(Item.CONFIGURE)) {
```

Comment:
This method looks very similar to the one above. Perhaps a helper function can be employed to reduce the redundant code.